gautomatch/protocols/protocol_gautomatch.py

```diff
@@ -338,18 +343,22 @@ def _pickMicrographList(self, micList, args):
             badCoords = self.inputBadCoords.get()
 
             if self.exclusive and badCoords:
-                fnCoords = os.path.join(self.getMicrographsDir(), '%s_rubbish.star'
+                fnCoords = pwutils.join(micPath, '%s_rubbish.star'
                                         % pwutils.removeBaseExt(micFn))
                 writeMicCoords(mic, badCoords.iterCoordinates(mic), fnCoords)
 
-        # We convert the input micrograph on demand if not in .mrc
         gautomatch.Plugin.runGautomatch(micFnList,
                                         self._getReferencesFn(),
-                                        self.getMicrographsDir(),
+                                        micPath,
                                         args,
                                         env=gautomatch.Plugin.getEnviron(),
                                         runJob=self.runJob)
 
+        # Move output from micPath (tmp) to extra
+        for p in [micPath + '/*.star', micPath + '/*.box']:
+            for f in (glob(p)):
+                pwutils.moveFile(f, self.getMicrographsDir())
```

Comment:
I think that it would be good to remove all temporary files once they has been parsed and the output is stored.